tests/gen/README.md

```diff
@@ -0,0 +1,154 @@
+# Automatic Documents Generation 
+
+`AutoGenerateDocs` is a highly versatile function designed for dynamic document generation, perfect for testing and simulation purposes. 
+It creates documents based on a specified schema, allowing for extensive customization of data generation. 
+
+The function generates documents adhering to a defined schema and it's configuration.
+It interprets the types and relationships within the schema to create realistic, interconnected data structures.
+
+### Demand Calculation:
+
+The function calculates the 'demand' or the number of documents to generate based on the configuration provided.
+For related types within the schema, it intelligently adjusts the number of generated documents to maintain consistency in relationships (one-to-one, one-to-many, etc.).
+
+In the absence of explicit demands, it deduces demands from the maximum required by related types or uses a default value if no relation-based demands are present.
+
+The error will be returned if the demand for documents can not be satisfied. 
+For example, a document expects at least 10 secondary documents, but the demand for secondary documents is 5.
+
+## Configuration
+
+There are two ways to configure the function:
+1. Directly within the schema using annotations
+2. Via options passed to the function
+
+Options take precedence over in-schema configurations.
+
+### In-schema Configuration:
+
+Field values can be configured directly within the schema using annotations after "#" (e.g., `# min: 1, max: 120` for an integer field).
+
+At the moment, the following value configurations are supported:
+- `min` and `max` for integer, float and relation fields. For relation fields, the values define the minimum and maximum number of related documents.
+- `len` for string fields
+
+Default value ranges are used when not explicitly set in the schema or via options.
+
+### Customization with Options:
+
+- `WithTypeDemand` and `WithTypeDemandRange` allow setting the specific number (or range) of documents for a given type.
+- `WithFieldRange` and `WithFieldLen` override in-schema configurations for field ranges and lengths.
+- `WithFieldGenerator` provides custom value generation logic for specific fields.
+- `WithRandomSeed` ensures deterministic output, useful for repeatable tests.
+
+## Examples
+
+### Basic Document Generation:
+
+```go
+schema := `
+type User {
+  name: String # len: 10
+  age: Int # min: 18, max: 50
+  verified: Boolean
+  rating: Float # min: 0.0, max: 5.0
+}`
+docs, _ := AutoGenerateDocs(schema, WithTypeDemand("User", 100))
```

Comment:
Would probably be something like:
`AutoGenerateDocs(definitions, WithTypeDemand("User", 100))`

You could then have a syntax sugar layer on top with roughly the current signature that converts and SDL to definitions before processing internally, you'd probably want to add a `client.Store` param or something for that though to let the prod code handle the conversion from SDL to definition.

Reply:
I'm tempted to see this as a todo, as there is a risk that this would become a major blocker and source of tech debt when we expose other means of creating schema/collections - we'll almost certainly want to include them in the test matrix, like we do for mutation and client types, and notable numbers of tests are using this (GQL SDL only) then they will inhibit full testing of that change.

We already exposed `db.CreateSchema(defintion)` briefly (earlier this year?), before hiding it (I cant remember why, I think it was a technical reason RE updating the GQL types), so I think there are decent odds of this being wanted again soon are reasonably high.